component/tests/test_build_component.py

```diff
@@ -0,0 +1,274 @@
+            self.comp_registry["component2"].__bases__,
+        )
+        self.assertEqual(
+            (Component3, self.comp_registry["component1"], self.comp_registry["base"]),
+            self.comp_registry["component3"].__bases__,
+        )
+        self.assertEqual(
+            (
+                Component4,
+                self.comp_registry["component2"],
+                self.comp_registry["component3"],
+                self.comp_registry["base"],
+            ),
+            self.comp_registry["component4"].__bases__,
+        )
+
+    def test_custom_build(self):
+        """Check that we can hook at the end of a Component build"""
+
+        class Component1(Component):
+            _name = "component1"
+
+            @classmethod
+            def _complete_component_build(cls):
+                # This method should be called after the Component
+                # is built, and before it is pushed in the registry
+                cls._build_done = True
+
+        Component1._build_component(self.comp_registry)
+        # we inspect that our custom build has been executed
+        self.assertTrue(self.comp_registry["component1"]._build_done)
+
+    def test_inherit_attrs(self):
+        """Check attributes inheritance of Components with _inherit"""
+
+        class Component1(Component):
+            _name = "component1"
+
+            msg = "ping"
+
+            def say(self):
+                return "foo"
+
+        class Component2(Component):
+            _name = "component2"
+            _inherit = "component1"
+
+            msg = "pong"
+
+            def say(self):
+                return super().say() + " bar"
+
+        Component1._build_component(self.comp_registry)
+        Component2._build_component(self.comp_registry)
+        # we initialize the components, normally we should pass
+        # an instance of WorkContext, but we don't need a real one
+        # for this test
+        component1 = self.comp_registry["component1"](mock.Mock())
+        component2 = self.comp_registry["component2"](mock.Mock())
+        self.assertEqual("ping", component1.msg)
+        self.assertEqual("pong", component2.msg)
+        self.assertEqual("foo", component1.say())
+        self.assertEqual("foo bar", component2.say())
+        return
+
+    def test_duplicate_component(self):
+        """Check that we can't have 2 components with the same name"""
+
+        class Component1(Component):
+            _name = "component1"
+
+        class Component2(Component):
+            _name = "component1"
+
+        Component1._build_component(self.comp_registry)
+        msg = "Component.*already exists.*"
+        with self.assertRaisesRegex(TypeError, msg):
+            Component2._build_component(self.comp_registry)
+
+    def test_no_parent(self):
+        """Ensure we can't _inherit a non-existent component"""
+
+        class Component1(Component):
```

Comment:
`component/tests/test_build_component.py:59: [R7980(consider-merging-classes-inherited), TestBuildComponent.test_inherit_bases.Component2] Consider merging classes inherited to "component1" from test_build_component.py:62, test_build_component.py:184, test_build_component.py:269.
`

@simahawk  pre commit complains for that as above. Possible to disable it somehow?

Reply:
in this case just customize the class w/ a silly method or attribute that is never used, eg:

```
class Component1(Component):
    def say_hello(self):
        """Dumb method."""
        return "hello"
```

Reply:
@simahawk  thanks, but I tried both this methods and params, to me it doesn't work.
`# pylint: disable=R7980` above affected classes maybe?

Reply:
Checked. Just do this: 
1. rollback your changes to address this
2. add this block after the file header

```py
# Copyright 2017 Camptocamp SA
# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html)

# Tell pylint to not bother us for all our fake component classes
# pylint: disable=consider-merging-classes-inherited
```